timeline.go

```diff
@@ -110,7 +111,7 @@ func (t *Timeline) deadline(at time.Time, now time.Time) context.Context {
 		if t.deadlines == nil {
 			t.deadlines = make(map[int64]deadline)
 		}
-		d = makeDeadline(background, expiration)
+		d = makeDeadline(background, jitterTime(expiration, time.Duration(r)))
```

Comment:
If I understand the code correctly, maybe we should do r/2 here?

Reply:
👍 or make the factor be a configurable property of the `Timeline`

Reply:
I don't think that's necessary. For example, if the timer resolution is 1s, the deadline is rounded down to the nearest second, then we add a random value between "0s" and "1s - 1ns" (due to the behavior of [rand.Int63n](https://golang.org/pkg/math/rand/#Rand.Int63n)), so I think it's already doing the right thing but maybe I'm missing something?

Reply:
My only thought during this was that it could in effect double the deadline in extreme cases.  So if your deadline is 5s, then as I read the code it could jitter it to 10s.

Reply:
Maybe I mis-read the code?

Reply:
Yeah, the case I'm thinking about is closer to misconfiguration but still would be nice to avoid.

That is, a Timeout or Deadline too close to the resolution.

Reply:
Not exactly, it may add up to 1 unit of resolution (which is configurable). It's already documented that the timer is as accurate as the configured resolution. If you use a 5s resolution for a short deadline you're doing it wrong, you should use a timer with a lower resolution.

The point is, applications already cannot predict the exact amount of time between now and when the timer triggers, and here we're adding a jitter value which doesn't modify the documented behavior. Distributing values across the full time window of the timer's resolution is the ideal scenario since it will best spread the triggers across services. If you ever need to change this it's likely that you just want a more accurate timer, not that you want to pack deadlines in a shorter time frame.